RELEASE.md

```diff
@@ -0,0 +1,57 @@
+Release
+=======
+
+Although not very tricky, the Ember Data release process does have a
+lot of manual steps. The following steps navigate us through
+some of the release gotchas and will hopefully result in a successful
+release.
+
+STEPS:
+------
+
+* generate changelog (`PRIOR_VERSION=v2.0.0 HEAD=release ./bin/changelog`)
+* prepend changelog output to `CHANGELOG.md`
+* edit changelog output to be as user-friendly as possible (drop [INTERNAL] changes, non-code changes, etc.)
+* Bump version in package.json
+* Git tag version
+* Do a production build. `rm -rf node_modules bower_components; npm install; bower install; npm run production`
+* Publish to Bower
+  * Commit built globals code to the https://github.com/components/ember-data repo
+  * `cp dist/globals/* ../components-ember-data/`
+* Publish Rubygems
+  * `gem build ember-data-source.gemspec`
+  * `gem push ember-data-source-2.4.0.beta.1.gem`
+* Publish to NPM
+  * `npm publish` or `npm publish --tag beta` or `npm publish --tag release-1-13`
+* Update the `/builds/` page on the website
+  * `cd ../website`
+  * Edit [lastRelease, futureVersion and date](https://github.com/emberjs/website/blob/master/source/javascripts/app/builds/app.js#L238-L241) values for the release channel we are releasing.
+  * Commit updated `javascripts/app/builds/app.js` file
+  * `rake deploy`
+* Website API docs (This step does not happen for beta releases)
+  * `cd ../website`
+  * `rake generate_ember_data_docs`
+  * Commit updated `data/data_api.yml` file
+  * `rake deploy`
+* Write a Release Blog Post (Does not happen for beta releases)
+  * Commits since last release: `git log --oneline release..beta | wc -l`.
+  * Contributors since last release: `git shortlog -s -n release...beta | wc -l`
+* Submit a Pull request to the https://github.com/ember-cli/ember-cli to update the version of Ember Data
+  * (per request by @rwjblue and is also a great idea to make upgrading/new apps easier)
```

Comment:
Maybe there should be an entry to update the version [in ember-twiddle](https://github.com/ember-cli/ember-twiddle/blob/master/app/services/dependency-resolver.js#L4) as well?